src/analyze/analyze.c

```diff
@@ -1297,19 +1300,49 @@ static int dump(int argc, char *argv[], void *userdata) {
                         "org.freedesktop.systemd1",
                         "/org/freedesktop/systemd1",
                         "org.freedesktop.systemd1.Manager",
-                        "Dump",
+                        "Dump2",
                         &error,
                         &reply,
                         NULL);
-        if (r < 0)
-                return log_error_errno(r, "Failed to issue method call: %s", bus_error_message(&error, r));
+        if (r < 0) {
+                const char *text = NULL;
+
+                /* fall back to Dump if either Dump2 or fd transfer is not supported */
+                if (!IN_SET(r, -EBADR, -EOPNOTSUPP))
+                        return log_error_errno(r, "Failed to issue method call: %s", bus_error_message(&error, r));
+
+                reply = sd_bus_message_unref(reply);
+                sd_bus_error_free(&error);
 
-        r = sd_bus_message_read(reply, "s", &text);
+                r = sd_bus_call_method(
+                                bus,
+                                "org.freedesktop.systemd1",
+                                "/org/freedesktop/systemd1",
+                                "org.freedesktop.systemd1.Manager",
+                                "Dump",
+                                &error,
+                                &reply,
+                                NULL);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to issue method call: %s", bus_error_message(&error, r));
+
+                r = sd_bus_message_read(reply, "s", &text);
+                if (r < 0)
+                        return bus_log_parse_error(r);
+
+                fputs(text, stdout);
+                return 0;
+        }
+
+        r = sd_bus_message_read(reply, "h", &remote_fd);
         if (r < 0)
                 return bus_log_parse_error(r);
 
-        fputs(text, stdout);
-        return 0;
+        fd = fcntl(remote_fd, F_DUPFD_CLOEXEC, 3);
+        if (fd < 0)
+                return log_error_errno(errno, "Failed to duplicate fd");
+
+        return copy_bytes(fd, STDOUT_FILENO, (uint64_t) -1, 0);
```

Comment:
hmm, just to be safe we should probably do a fflush(stdout) here first, to minimize the chance that our low-level writing to fd 1 and stdio interefere.